File: javascript/sieve/sieve.js
var Sieve = function(limit) {
	this.limit = limit;
	this.primes = this.findPrimes();
};

Sieve.prototype.generateSequence = function() {
	var sequence = {};
	for (var i = 2; i <= this.limit; ++i) {
		sequence[i] = false;
	}
	
	return sequence;
};

Sieve.prototype.findPrimes = function() {
	var sequence = this.generateSequence();
	
	for (var i = 2; i <= this.limit; ++i) {
		// Jumps to the next iteration of the loop in case the current iteration
		// (current `i`) is a number of the sequence we already marked.
		if (true === sequence[i]) {
			continue;
		}
		
		for (var j = i + 1; j <= this.limit; ++j) {
			if (0 === (j % i)) {
				sequence[j] = true;
			}
		}
		
	}
	
	return this.objToArr(sequence);
	
};

Sieve.prototype.objToArr = function(obj) {
	var arr = [];
	for (var key in obj) {
		if (false === obj[key]){
			arr.push(parseInt(key));
		}
	}
	
	return arr;
};

module.exports = Sieve;

File: python/sieve/sieve.py
def indexestoarr(arr):
	output = []
	for i in range(len(arr)):
		if (False == arr[i]):
			output.append(i)
	
	return output

def sieve(limit):
	sequence = [False] * (limit + 1)
	
	sequence[0] = True
	sequence[1] = True
	
	for i in range(2, limit + 1):
		if (True == sequence[i]):
			continue
		
		for j in range(i + 1, limit + 1):
			if (0 == (j % i)):
				sequence[j] = True
	
	# TODO: Inspect this kind of syntax
	# return [i for i, j in enumerate(sequence) if not j]
	return indexestoarr(sequence)